File: java/series/src/main/java/Series.java
import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.IntStream;

public class Series {
    private String numStr;

    public Series(String number) {
        this.numStr = number;
    }

    public List<String> slices(int n) {
        if (numStr.length() < n)
            throw new IllegalArgumentException("Slice size is too big.");
        if (n < 1)
            throw new IllegalArgumentException("Slice size is too small.");
        
        return IntStream.rangeClosed(0, numStr.length() - n)
                .mapToObj(i -> numStr.substring(i, i + n))
                .collect(Collectors.toList());
      
    }
}

File: swift/series/Sources/Series/Series.swift
struct Series {
    var input: [Int]

    init(_ input: String) {
        self.input = input.map { Int(String($0))! }
    }

    func slices(_ number: Int) -> [[Int]] {
        guard number <= input.count else {
            return []
        }
        return (0...input.count - number).map {
            Array(input[$0..<($0 + number)])
        }
    }
}
